ingest/src/main/java/com/microsoft/azure/kusto/ingest/StreamingIngestClient.java

```diff
@@ -147,25 +147,17 @@ public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, Inges
         return new IngestionStatusResult(ingestionStatus);
     }
 
-    private String getFormat(IngestionProperties ingestionProperties) {
-        String format = ingestionProperties.getDataFormat();
-        if (format == null) {
-            return "csv";
-        }
-        return format;
-    }
-
-    private String getMappingReference(IngestionProperties ingestionProperties, String format) throws IngestionClientException {
+    private String getMappingReference(IngestionProperties ingestionProperties, IngestionProperties.DataFormat dataFormat) throws IngestionClientException {
         IngestionMapping ingestionMapping = ingestionProperties.getIngestionMapping();
         String mappingReference = ingestionMapping.getIngestionMappingReference();
         IngestionMapping.IngestionMappingKind ingestionMappingKind = ingestionMapping.getIngestionMappingKind();
-        if (IngestionMapping.mappingRequiredFormats.contains(format)) {
+        if (dataFormat.isMappingRequired()) {
             String message = null;
-            if (!format.equalsIgnoreCase(ingestionMappingKind.name())) {
-                message = String.format("Wrong ingestion mapping for format %s, found %s mapping kind.", format, ingestionMappingKind.name());
+            if (!dataFormat.getIngestionMappingKind().equals(ingestionMappingKind)) {
```

Comment:
This was a bug, expecting identical DF/IMK (in reality they have a many-to-one relationship, via a mapping)